readthedocs/search/faceted_search.py

```diff
@@ -233,22 +238,19 @@ def query(self, search, query):
             fields=self.fields,
         )
 
-        # run bool query with should, so it returns result where either of the query matches
+        # Run bool query with should, so it returns result where either of the query matches.
         bool_query = Bool(should=queries)
-        search = search.query(bool_query)
-        return search
 
+        # Filter by project slugs.
+        if self.projects and isinstance(self.projects, list):
```

Comment:
What if it isn't a list? This seems fragile, and we should either error or force it into a list.

Reply:
Wasn't sure how to handle this one, I had two ideas:

1) Override the init method on both children classes to only support list in one and list and dict in the other one
2) Or just error in this class

Maybe I'll go with 2